### ISSUE TITLE
[Task]: Remove Flink 1.14 and cleanup

### ISSUE THREAD

**Issue by je-ik** [apache/beam#31112]
### What needs to happen?

After #30789 we can remove support for Flink 1.14 and review code to do cleanup of unneeded version specific code in `runners/flink/<version>/src`.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner
